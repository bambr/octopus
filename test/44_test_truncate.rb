#!/usr/bin/ruby

$: << File.dirname($0) + '/lib'
require 'run_env'

RunEnv.env_eval do
	start

	c = connect

	#
	# Проверяем работу LUA-функции полного удаления данных
	#
	# Данная функция работает только в рамках транзакции изменения
	# данных, которая автоматически запускается при выполнении LUA-функций
	# с использованием BOX/IProto-интерфейса
	#
	c.insert [1, 2, "abc", "def"]
	c.select 1, 2
	c.lua "user_proc.truncate", "0"
	c.select 1, 2

	#
	# Проверяем работу специальной команды полного удаления данных,
	# которая относится к классу BOX/IProto-команд модификации
	# мета-информации
	#
	c.insert [2, 3, "klm", "xyz"]
	c.select 1, 2
	c.truncate
	c.select 1, 2
end
