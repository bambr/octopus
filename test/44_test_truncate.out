# box.insert([1, 2, "abc", "def"])
1

# box.select(1, 2)
[["\x01\x00\x00\x00", "\x02\x00\x00\x00", "abc", "def"]]

# box.lua("user_proc.truncate", "0")
[]

# box.select(1, 2)
[]

# box.insert([2, 3, "klm", "xyz"])
1

# box.select(1, 2)
[["\x02\x00\x00\x00", "\x03\x00\x00\x00", "klm", "xyz"]]

# box.truncate()
:success

# box.select(1, 2)
[]

